File: algo/src/test/java/com/hartwig/actin/algo/evaluation/priortumor/HasHistoryOfSecondMalignancyWithDOIDTest.java
package com.hartwig.actin.algo.evaluation.priortumor;

import static com.hartwig.actin.algo.evaluation.EvaluationAssert.assertEvaluation;

import java.util.List;

import com.google.common.collect.Lists;
import com.hartwig.actin.algo.datamodel.EvaluationResult;
import com.hartwig.actin.clinical.datamodel.PriorSecondPrimary;
import com.hartwig.actin.doid.DoidModel;
import com.hartwig.actin.doid.TestDoidModelFactory;

import org.junit.Test;

public class HasHistoryOfSecondMalignancyWithDOIDTest {

    @Test
    public void canEvaluate() {
        DoidModel doidModel = TestDoidModelFactory.createWithOneParentChild("100", "200");
        HasHistoryOfSecondMalignancyWithDOID function = new HasHistoryOfSecondMalignancyWithDOID(doidModel, "100");

        // No prior tumors.
        List<PriorSecondPrimary> priorTumors = Lists.newArrayList();
        assertEvaluation(EvaluationResult.FAIL, function.evaluate(PriorTumorTestFactory.withPriorSecondPrimaries(priorTumors)));

        // Wrong doid
        priorTumors.add(PriorTumorTestFactory.builder().addDoids("300").build());
        assertEvaluation(EvaluationResult.FAIL, function.evaluate(PriorTumorTestFactory.withPriorSecondPrimaries(priorTumors)));

        // Right doid
        priorTumors.add(PriorTumorTestFactory.builder().addDoids("200").build());
        assertEvaluation(EvaluationResult.PASS, function.evaluate(PriorTumorTestFactory.withPriorSecondPrimaries(priorTumors)));
    }
}

File: algo/src/test/kotlin/com/hartwig/actin/algo/evaluation/priortumor/HasHistoryOfSecondMalignancyWithDoidTest.kt
package com.hartwig.actin.algo.evaluation.priortumor

import com.hartwig.actin.algo.evaluation.EvaluationAssert.assertEvaluation
import com.hartwig.actin.datamodel.algo.EvaluationResult
import com.hartwig.actin.datamodel.clinical.PriorPrimary
import com.hartwig.actin.doid.TestDoidModelFactory
import org.junit.jupiter.api.Test

class HasHistoryOfSecondMalignancyWithDoidTest {

    @Test
    fun canEvaluate() {
        val doidModel = TestDoidModelFactory.createWithOneParentChild("1", "2").copy(termForDoidMap = mapOf("1" to "breast cancer"))
        val function = HasHistoryOfSecondMalignancyWithDoid(doidModel, "1")

        // No prior tumors.
        val priorTumors: MutableList<PriorPrimary> = mutableListOf()
        assertEvaluation(
            EvaluationResult.FAIL,
            function.evaluate(PriorTumorTestFactory.withPriorPrimaries(priorTumors)),
            "No history of previous malignancy belonging to breast cancer"
        )

        // Wrong doid
        priorTumors.add(PriorTumorTestFactory.priorPrimary(doid = "3"))
        assertEvaluation(
            EvaluationResult.FAIL,
            function.evaluate(PriorTumorTestFactory.withPriorPrimaries(priorTumors)),
            "No history of previous malignancy belonging to breast cancer"
        )

        // Right doid
        priorTumors.add(PriorTumorTestFactory.priorPrimary(doid = "2"))
        assertEvaluation(
            EvaluationResult.PASS,
            function.evaluate(PriorTumorTestFactory.withPriorPrimaries(priorTumors)),
            "History of previous malignancy belonging to breast cancer"
        )
    }
}